### PR TITLE
hexgui: init at unstable-2022-5-30

### PR DESCRIPTION
**Pull request by ursi**
###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by Kranzes** [comment]
1v1 me in hex

**Comment by ursi** [comment]
if I win, you have to merge the PR

**Comment by Kranzes** [comment]
> if I win, you have to merge the PR

Looks like this PR is never getting merged then 😆

**Comment by ursi** [comment]
@SuperSandro2000 I believe this is ready to be merged.

**Comment by ursi** [comment]
(forgot the "at" in the commit message)

File: pkgs/games/hexgui/default.nix [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+
+  src = fetchFromGitHub {
+    owner = "selinger";
+    repo = "hexgui";
+    rev = "d94ce1d35a22dad28d3e7def4d28e6bebd54da9d";
+    hash = "sha256-1MroFH2JSEZHFigcsw1+xyHJWEnHTvHmRPVirUgwM6I=";
+  };
+
+  nativeBuildInputs = [ ant jdk makeWrapper ];
+  buildPhase = "ant";
```
Comment (SuperSandro2000) [inline pkgs/games/hexgui/default.nix:20]:
```suggestion
  buildPhase = ''
    ant
  '';
```

File: pkgs/games/hexgui/default.nix [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+  version = "unstable-2022-5-30";
+
+  src = fetchFromGitHub {
+    owner = "selinger";
+    repo = "hexgui";
+    rev = "d94ce1d35a22dad28d3e7def4d28e6bebd54da9d";
+    hash = "sha256-1MroFH2JSEZHFigcsw1+xyHJWEnHTvHmRPVirUgwM6I=";
+  };
+
+  nativeBuildInputs = [ ant jdk makeWrapper ];
```
Comment (SuperSandro2000) [inline pkgs/games/hexgui/default.nix:19]: Why do we have jdk in nativeBuildInputs?

File: pkgs/games/hexgui/default.nix [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+  buildPhase = "ant";
+
+  installPhase = ''
+    mkdir $out;
```
Comment (SuperSandro2000) [inline pkgs/games/hexgui/default.nix:23]:
```suggestion
    mkdir $out
```

File: pkgs/games/hexgui/default.nix [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+    cp -r bin lib $out
```
Comment (SuperSandro2000) [inline pkgs/games/hexgui/default.nix:24]:
```suggestion
    mv bin lib $out
```
mv should be faster on same fs

File: pkgs/games/hexgui/default.nix [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+    wrapProgram $out/bin/hexgui --prefix PATH : ${jdk}/bin
+  '';
+
+  meta = {
+    description = "HexGui";
+    longDescription = "GUI for the board game Hex (and Y)";
```
Comment (SuperSandro2000) [inline pkgs/games/hexgui/default.nix:30]:
```suggestion
    description = "GUI for the board game Hex (and Y)";
    homepage = "https://github.com/selinger/hexgui";
```

File: pkgs/games/hexgui/default.nix [hunk 6]
```diff
@@ -0,0 +1,34 @@
...
+    hash = "sha256-1MroFH2JSEZHFigcsw1+xyHJWEnHTvHmRPVirUgwM6I=";
+  };
+
+  nativeBuildInputs = [ ant jdk makeWrapper ];
+  buildPhase = "ant";
+
+  installPhase = ''
+    mkdir $out;
+    cp -r bin lib $out
+    wrapProgram $out/bin/hexgui --prefix PATH : ${jdk}/bin
```
Comment (SuperSandro2000) [inline pkgs/games/hexgui/default.nix:25]: Why do we need to wrap it with jdk? Does it exec to that?

```suggestion
    wrapProgram $out/bin/hexgui --prefix PATH : ${lib.makeBinPath [ jdk ]}
```

File: maintainers/maintainer-list.nix [hunk 7]
```diff
@@ -14236,6 +14236,12 @@
     githubId = 32751441;
     name = "urlordjames";
   };
+  ursi = {
```
Comment (SuperSandro2000) [inline maintainers/maintainer-list.nix:14239]: Please put this into a separate commit with the message ``maintainers: add NAME`` and sort it alphabetically.

**Review by SuperSandro2000 — CHANGES_REQUESTED** [review]

File: pkgs/games/hexgui/default.nix (hunk 2 above)
Comment (Kranzes) [reply pkgs/games/hexgui/default.nix:19]: Needed for `ant` I believe.

Comment (ursi) [reply pkgs/games/hexgui/default.nix:19]: I just copied this https://nixos.org/manual/nixpkgs/stable/#sec-language-java

File: pkgs/games/hexgui/default.nix (hunk 5 above)
Comment (ursi) [reply pkgs/games/hexgui/default.nix:30]: So the fact that the actual name of the app is "HexGui" shouldn't be documented anywhere?

**Comment by ursi** [comment]
(I will squash these, just committing them here to make it easier)

File: pkgs/games/hexgui/default.nix (hunk 5 above)
Comment (ursi) [reply pkgs/games/hexgui/default.nix:32]: I was going off of this https://github.com/NixOS/nixpkgs/blob/16f4e04658c2ab10114545af2f39db17d51bd1bd/pkgs/applications/version-management/gitlab/default.nix#L220-L221

Comment (SuperSandro2000) [reply pkgs/games/hexgui/default.nix:32]: meta.description should be short but not that short.

> So the fact that the actual name of the app is "HexGui" shouldn't be documented anywhere?

Thats just a casing difference.
